### PR TITLE
Bump PyYAML upper-bound 

### PR DESCRIPTION
**Pull request by datajoely**
## Description
Main rationale of achieving [pyodide](https://github.com/pyodide/pyodide) compatibility documented in [this issue](https://github.com/kedro-org/kedro/issues/1541).

## Development notes
- Dropping lower bound will get unit tests to pass, but e2e tests would fail because telemetry requires the old bounds.
- Kept the existing lower bound for compatibility, only major impact change is the dropping of 2.7 support from YAML side.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1545"></a>

### CODE REVIEW

**Review by MerelTheisenQB — APPROVED** [review]
LGTM! 👍 Thanks for looking into how we can support `pyodide`.

**Review by SajidAlamQB — APPROVED** [review]
Thank you. 👍

File: RELEASE.md [hunk 1]
```diff
@@ -8,11 +8,17 @@
...
+
+# Release 0.18.2
+
 # Upcoming Release 0.18.2
 
 ## Major features and improvements
 
 ## Bug fixes and other changes
+
+* Bumped `pyyaml` upper-bound to 7.0.0. Version 6.0.0 of PyYAML dropoed Python 2.7 support and relaxing this makes Kedro compatible with the [pyodide](https://pyodide.org/en/stable/usage/loading-packages.html#micropip) stack.
```
Comment (AntonyMilneQB) [inline RELEASE.md:21]:
```suggestion
* Bumped `pyyaml` upper-bound to make Kedro compatible with the [pyodide](https://pyodide.org/en/stable/usage/loading-packages.html#micropip) stack.
```
Just to keep the release notes snappy 😀 I don't think anyone cares about Python 2.7 support. Probably most people would think it's a typo for 3.7...

**Review by AntonyMilneQB — COMMENTED** [review]
Don't forget the yaml dataset requirement in setup.py also!

**Pull request merged**